src/Filesystem/Replicate.php

```diff
@@ -214,16 +214,48 @@ public function getAdapterClassNames()
 
     public function createFile($key, Filesystem $filesystem)
     {
+        \assert($this->primary instanceof FileFactory);
```

Comment:
Why this assert is needed ? Shouldnt we change some type hint or throw an exception instead ?

Reply:
I think I solved on master doing an IF instead.

Reply:
I backported the `if` from `master` only for the primary adapter, which is the current behavior.